cli/azd/pkg/environment/azdcontext/azdcontext.go

```diff
@@ -68,14 +68,26 @@ func (c *AzdContext) GetDefaultEnvironmentName() (string, error) {
 	return config.DefaultEnvironment, nil
 }
 
-func (c *AzdContext) SetDefaultEnvironmentName(name string) error {
+// ProjectState represents the state of the project.
+type ProjectState struct {
+	DefaultEnvironment string
+}
+
+// SetProjectState persists the state of the project to the file system, like the default environment.
+func (c *AzdContext) SetProjectState(state ProjectState) error {
 	path := filepath.Join(c.EnvironmentDirectory(), ConfigFileName)
 	config := configFile{
 		Version:            ConfigFileVersion,
-		DefaultEnvironment: name,
+		DefaultEnvironment: state.DefaultEnvironment,
+	}
+
+	if err := writeConfig(path, config); err != nil {
+		return err
 	}
 
-	return writeConfig(path, config)
+	// make sure to ignore the environment directory
+	path = filepath.Join(c.EnvironmentDirectory(), ".gitignore")
+	return os.WriteFile(path, []byte("# .azure is not intended to be committed\n*"), osutil.PermissionFile)
```

Comment:
Should we do this write only if `.gitignore` doesn't already exist, to allow the user to change the contents if they wished?

Reply:
Could we also append .azure, if `.gitignore` already exists?

Reply:
> Should we do this write only if `.gitignore` doesn't already exist, to allow the user to change the contents if they wished?

This seems to be the safest thing to do - and if they really wanted to commit something they could add perform a force add to override the .gitignore.